src/argilla/client/feedback/dataset/remote/dataset.py

```diff
@@ -482,6 +515,27 @@ def add_records(
         """
         self._records.add(records=records, show_progress=show_progress)
 
+    @allowed_for_roles(roles=[UserRole.owner, UserRole.admin])
+    def find_similar_records(
+        self,
+        vector_name: str,
+        value: Optional[List[float]] = None,
+        record: Optional[RemoteFeedbackRecord] = None,
+        max_results: int = 50,
+    ) -> List[Tuple[RemoteFeedbackRecord, float]]:
+        """Finds similar records to the given record in the dataset based on the given vector.
+
+        Args:
+            vector_name: a vector name to use for searching by similarity.
+            value: an optional vector value to be used for searching by similarity.
+            record: an optional record to be used for searching by similarity.
+            max_results: the maximum number of results for the search.
```

Comment:
```suggestion
            value: an optional vector value to be used for searching by similarity. Defaults to None.
            record: an optional record to be used for searching by similarity. Defaults to None.
            max_results: the maximum number of results for the search. Defaults to 50.
```

Reply:
And I'd also add a note that one of `value` or `record` needs to be provided, and not both.